Sources/FoundationInternationalization/Calendar/Calendar_ICU.swift

```diff
@@ -1158,8 +1158,7 @@ internal final class _Calendar: Equatable, @unchecked Sendable {
             var cease: UInt32?
 
             for i in 0..<7 {
-                var status = U_ZERO_ERROR
-                weekdayTypes[i] = ucal_getDayOfWeekType(ucalendar, UCalendarDaysOfWeek(rawValue: weekdaysIndex[i]), &status)
+                weekdaysIndex[i] = UInt32(_ucal_getDayOfWeekType(ucalendar, CInt(weekdaysIndex[i])))
```

Comment:
We lost the status here, but I guess we didn't check the result anyway?

Reply:
I think I remember spelunking into ICU for this and verifying it was never set, but it's probably worth double checking my work there.

Reply:
Correct, and this was intentional.  If we care about the status, we should actually mark the function as `throws` and surface the status as an error.

Reply:
I went back and forth on this for a bit, and in the end decided that I wouldn't try to Swiftify the ICU APIs, at least not here and now. Instead I added a `check` function to the error type and make it an explicit step after calling them.

Reply:
It _can_ return a `U_ILLEGAL_ARGUMENT_ERROR` as a status in the case that `dayOfWeek` is invalid.  But that we should be caught by the compiler, so 🤷‍♂️.